reconcile/utils/terrascript_client.py

```diff
@@ -4451,3 +4465,190 @@ def populate_tf_resource_route53_zone(self, resource, namespace_info):
+        account, identifier, common_values, output_prefix, \
+            output_resource_name, annotations = \
+            self.init_values(resource, namespace_info)
+        tf_resources = []
+        self.init_common_outputs(tf_resources, namespace_info, output_prefix,
+                                output_resource_name, annotations)
+
+        # FIXME
+        # These should be sourced from ai provider values
+        sms_role_ext_id = "foobar"
+        organization_name = "rosa-fedramp"
+        cognito_user_pool_domain = "cognito.fedramp.devshift.net"
+
+        sms_role_policy = {
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Effect": "Allow",
+                    "Action": [
+                        "sts:AssumeRole",
+                    ],
+                    "Principal": {
+                        "Service": "cognito-idp.amazonaws.com",
+                    },
+                    "Condition": {
+                        "StringEquals": {
+                            "sts:ExternalId": sms_role_ext_id
+                        }
+                    }
+                },
+                {
+                    "Effect": "Allow",
+                    "Action": [
+                        "sns:Publish",
+                    ],
+                    "Resource": "*",
+                }
+            ]
+        }
+
+        sms_iam_role_resource = aws_iam_role(
+            "sms_role",
+            name=f'{organization_name}-SMS',
+            description="role for applicant cognito, send sms",
+            assume_role_policy=sms_role_policy,
+            force_detach_policies=False,
+            max_session_duration=3600,
+            path="/service-role/",
+        )
+        tf_resources.append(sms_iam_role_resource)
+
+        # https://registry.terraform.io/providers/hashicorp/aws/3.60.0/docs/resources/cognito_identity_provider
+
+        # What is the difference between values and self.get_values?
+        # How do they relate to default files vs supplied parameters?
+        # I'm trying to use RDS as an example and I am getting confused with application of pg_values (parameter group)
+
+        pool_args = values.get("user_pool_properties", None)
+        pool_client_args = values.get("user_pool_client_properties", None)
+        
+        if pool_args and pool_client_args:
+            cognito_user_pool_resource = aws_cognito_user_pool(
+                "pool",
+                name=f'{organization_name}-pool',
+                **pool_args
+            )
+
+            tf_resources.append(cognito_user_pool_resource)
+            # How does terraform DAG work in terrascript?
+            #
+            # For instance, I need to reference the ID of the user pool defined above in the user_pool_domain resource
+            # below. Does terrascript automatically respect the terraform DAG and supply this value
```

Comment:
"Does terrascript automatically respect the terraform DAG and supply this value"

I believe the answer is basically yes. Terrascript doesn't do anything special, nor does it need to do. It just produces raw JSON format terraform files (which are equivalent to the more common HCL format terraform files), and terraform reads these JSON format files directly. At that point, it is terraform that is automatically handling the deciding on the most optimal order of resource provisioning, exactly the same as if you were providing Terraform with a normal HCL format terraform file.